### PR TITLE
fix(zc): remove ladder during warp

### PR DESCRIPTION
**Pull request by Jambu2024**

### CODE REVIEW

File: src/zc/hero.cpp [hunk 1]
```diff
@@ -25490,7 +25490,8 @@ bool HeroClass::dowarp(int32_t type, int32_t index, int32_t warpsfx)
 	// Drawing commands probably shouldn't carry over...
 	if ( !get_qr(qr_SCRIPTDRAWSINWARPS) )
 		script_drawing_commands.Clear();
-	
+
+	reset_ladder();
```
Comment (connorjclark) [inline src/zc/hero.cpp]: I don't know exactly where this should be moved to instead, but as it is here, it seems the ladder can go away too soon. for example, if the player warps while on a ladder, then the ladder goes away as the screen transitions.

https://zc-replay-compare-11639869472.surge.sh/

`nargads_trail_crystal_crusades/nargads_trail_crystal_crusades_07_of_24.zplay - 23881`
image
image

the ladder should be present during the first part of the screen transition, but be gone during the second, yeah?

Comment (Jambu2024) [reply src/zc/hero.cpp:25494]: oh raft not behaving that's interesting.

**Comment by Jambu2024** [comment]
@connorjclark 
their are no changes to the raft so i think we can merge this through now that the checks are passing.

File: src/zc/hero.cpp [hunk 2]
```diff
@@ -25461,36 +25461,40 @@ bool HeroClass::HasHeavyBoots()
...
+
 	// Drawing commands probably shouldn't carry over...
-	if ( !get_qr(qr_SCRIPTDRAWSINWARPS) )
+	if (!get_qr(qr_SCRIPTDRAWSINWARPS))
 		script_drawing_commands.Clear();
-	
+
+	if (replay_version_check(35))
+	{
+		reset_ladder();
```
Comment (connorjclark) [inline src/zc/hero.cpp:25498]:
```suggestion
		// TODO: Would be better to remove a bit later, after fading out the current screen. For now, do this here.
		// https://github.com/ZQuestClassic/ZQuestClassic/pull/1017#discussion_r1828880117
		reset_ladder();
```

**Review by connorjclark — APPROVED** [review]

**Pull request merged**